bibliopixel/drivers/SPI/driver.py

```diff
@@ -0,0 +1,42 @@
+from .. channel_order import ChannelOrder
+from .. driver_base import DriverBase
+from ... util.enum import resolve_enum
+from . import interfaces
+from enum import IntEnum
+# SubDriver imports
+from . APA102 import APA102
+from . LPD8806 import LPD8806
+from . WS281X import WS281XSPI
+from . WS2801 import WS2801
+
+
+class SPI_LEDTYPE(IntEnum):
+    APA102 = 0
+    DOTSTAR = 0
+    WS2801 = 1
+    WS281X = 2
+    WS2812 = 2
+    WS2812B = 2
+    NEOPIXEL = 2
+    LPD8806 = 3
+
+
+SPI_DRIVERS = [
+    APA102,
+    WS2801,
+    WS281XSPI,
+    LPD8806
+]
+
+
+class SPI(DriverBase):
```

Comment:
In PixelWeb I have a tool that will scan a file and generate a MANIFEST template with as much information as it can reflect out of the file. One of the things I use for that is if an object derives from DriverBase I know it's a driver and handle it appropriately. I realize we no longer support PixelWeb or the MANIFEST vars, but the intent is to bring all that back in some form or another. So, everything that is a driver should inherit from DriverBase. Granted, yes, we could solve it with MANIFEST as well but I'd much rather do as much of that via automatic detection and inheritance seemed the simplest way.